core/src/main/java/org/apache/iceberg/SnapshotScan.java

```diff
@@ -79,6 +79,26 @@ protected ScanMetrics scanMetrics() {
     return scanMetrics;
   }
 
+  protected Map<Integer, PartitionSpec> specs() {
+    Map<Integer, PartitionSpec> specs = table().specs();
+    // requires latest schema
+    if (!useSnapshotSchema()
+        || snapshotId() == null
+        || table().currentSnapshot() == null
+        || snapshotId().equals(table().currentSnapshot().snapshotId())) {
+      return specs;
+    }
+
+    // this is a time travel request
+    Schema snapshotSchema = tableSchema();
+    Map<Integer, PartitionSpec> newSpecs = Maps.newHashMapWithExpectedSize(specs.size());
```

Comment:
Are we returning mutable map intentionally? How about using `ImmutableMap.toImmutableMap` or `Collections.unmodifiableMap`?

Reply:
We could use the ImmutableMap Builder  to avoid the extra variable

Reply:
Thanks @ebyhr @pvary 
I updated to use `Collections.unmodifiableMap(newSpecs)` for consistency with other usages

Reply:
Thanks.
Interesting part of creation of map:
```
 g grep Maps.newHashMap | wc -l         
    1485
 ```